Runtimes/Supplemental/cmake/modules/FindSwiftOverlay.cmake

```
@@ -0,0 +1,236 @@
+    set(SwiftOverlay_USE_STATIC_LIBS ON)
+  endif()
+endif()
+
+if(SwiftOverlay_DIR)
+  find_package(SwiftOverlay CONFIG)
+  return()
+endif()
+
+if(APPLE)
+  list(APPEND OVERLAY_TARGET_NAMES "swiftDarwin")
+  # Look in the SDK
+  list(APPEND swiftDarwin_INCLUDE_DIR_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  list(APPEND swiftDarwin_LIBRARY_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  # When building for Apple platforms, swiftDarwin always comes from within the
+  # SDK as a tbd for a shared library in the shared cache.
+  list(APPEND swiftDarwin_NAMES libswiftDarwin.tbd)
+  set(swiftDarwin_MODULE_NAME "Darwin.swiftmodule")
+elseif(LINUX)
+  #ToDo(swiftlang/swift/issues/83014): Handle the static MUSL SDK case
+  list(APPEND OVERLAY_TARGET_NAMES "swiftGlibc")
+
+  # If the swiftGlibc_DIR_FLAG is specified, look there instead. The cmake-generated
+  # config file is more accurate
+  if(swiftGlibc_DIR)
+    if(swiftGlibc_FIND_REQUIRED)
+      list(APPEND args REQUIRED)
+    endif()
+    if(swiftGlibc_FIND_QUIETLY)
+      list(APPEND args QUIET)
+    endif()
+    find_package(swiftGlibc NO_MODULE ${args})
+  # Look in the SDK
+  else()
+    if (SwiftOverlay_USE_STATIC_LIBS)
+      list(APPEND swiftGlibc_INCLUDE_DIR_HINTS
+        "${Swift_SDKROOT}/usr/lib/swift_static/linux")
+      list(APPEND swiftGlibc_LIBRARY_HINTS
+        "${Swift_SDKROOT}/usr/lib/swift_static/linux")
+      list(APPEND swiftGlibc_NAMES libswiftGlibc.a)
+    else()
+      list(APPEND swiftGlibc_INCLUDE_DIR_HINTS
+        "${Swift_SDKROOT}/usr/lib/swift/linux/")
+      list(APPEND swiftGlibc_LIBRARY_HINTS
+        "${Swift_SDKROOT}/usr/lib/swift/linux")
+      list(APPEND swiftGlibc_NAMES libswiftGlibc.so)
+    endif()
+    set(swiftGlibc_MODULE_NAME "Glibc.swiftmodule")
+  endif()
+elseif(WIN32)
+  list(APPEND OVERLAY_TARGET_NAMES "swiftWinSDK")
+  list(APPEND OVERLAY_TARGET_NAMES "swiftCRT")
+
+  list(APPEND swiftWinSDK_INCLUDE_DIR_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/windows"
+    "$ENV{SDKROOT}/usr/lib/swift/windows")
+  list(APPEND swiftWinSDK_LIBRARY_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "${Swift_SDKROOT}/usr/lib/swift"
+    "$ENV{SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "$ENV{SDKROOT}/usr/lib/swift")
+  list(APPEND swiftWinSDK_NAMES swiftWinSDK.lib)
+  list(APPEND swiftWinSDK_MODULE_NAME "WinSDK.swiftmodule")
+
+  list(APPEND swiftCRT_INCLUDE_DIR_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/windows"
+    "$ENV{SDKROOT}/usr/lib/swift/windows")
+  list(APPEND swiftCRT_LIBRARY_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "${Swift_SDKROOT}/usr/lib/swift"
+    "$ENV{SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "$ENV{SDKROOT}/usr/lib/swift")
+  list(APPEND swiftCRT_NAMES swiftCRT.lib)
```

We need to change the name whether we are looking for dynamic or static libraries
(please double check with what we did for `FindSwiftCore`)

```suggestion
  if(SwiftOverlay_USE_STATIC_LIBS)
    list(APPEND swiftCRT_NAMES libswiftCRT.lib)
  else()
    list(APPEND swiftCRT_NAMES swiftCRT.lib)
  endif()
```